lib/events_processor.rb

```diff
@@ -17,8 +17,9 @@ def process
       Rails.logger.error errors unless errors.empty?
     end
 
-    def retrieve_reviews
-      Event.where('data ?| array[:keys]', keys: Event::TYPES).find_each.lazy.each do |event|
+    def retrieve_entity_records(entity)
+      entity = Event::TYPES if entity == 'all'
+      Event.where('data ?| array[:keys]', keys: entity).find_each.lazy.each do |event|
```

Comment:
What do you think of...

```suggestion
    def retrieve_entity_records(types)
      types = Event::TYPES if types == 'all'
      Event.where('data ?| array[:keys]', keys: types).find_each.lazy.each do |event|
```